Robust/ViewModel/ShoppingCartViewModel.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Robust.Model.Product;
+using Robust.Service;
+using Robust.ViewModel.RelayCommands;
+using Robust.MVVM.ViewModelBase;
+
+namespace Robust.ViewModel
+{
+    public class ShoppingCartViewModel : ViewModelBase
+    {
+        private IWindowService _windowService;
+        //This list is used to store the products currently in the shopping cart.
+        public ObservableCollection<Product> ShoppingCartList { get; set; }
+        
+        //The UI doesn't update the total price properly - this needs to be fixed.
+        private decimal _totalPrice;
+
+        public decimal TotalPrice
+        {
+            get { return _totalPrice; }
+            set 
+            { 
+                _totalPrice = value;
+                OnPropertyChanged(nameof(TotalPrice));
+            }
+        }
+
+        private Product _selectedProduct;
+
+        public Product SelectedProduct
+        {
+            get { return _selectedProduct; }
+            set
+            {
+                _selectedProduct = value;
+                OnPropertyChanged();
+            }
+        }
+
+        //We should maybe include a property connected to Product called Quantity. Here we use StockQuantity as a measure of how many items of each Product there are.
+        public void CalculateTotalPrice()
+        {
+            TotalPrice = 0;
+
+            foreach (Product item in ShoppingCartList)
+            {
+                TotalPrice = TotalPrice + item.Price * item.StockQuantity;
+            }
+        }
+
+        public RelayCommand ToCheckoutCmd => new RelayCommand(ShowCheckout, CanShowCheckout);
+
+        private void ShowCheckout()
+        {
+            _windowService.ShowDialog(ShoppingCartList);
+        }
+
+        private bool CanShowCheckout()
+        {
+            return ShoppingCartList != null && ShoppingCartList.Count > 0;
+        }
+
+        public RelayCommand DeleteSelectedProductCmd => new RelayCommand(DeleteSelectedProduct, CanDeleteSelectedProduct);
+
+        private void DeleteSelectedProduct()
+        {
+            ShoppingCartList.Remove(SelectedProduct);
+
+            TotalPrice = 0;
+        }
+
+        private bool CanDeleteSelectedProduct()
+        {
+            return SelectedProduct != null;
+        }
```

Comment:
Når en metode er relativ simpel såsom denne kan vi lave det indtil en lambda expression som gør det mere læsbarligt:

```suggestion
        private bool CanDeleteSelectedProduct() => SelectedProduct != null;
```